### PR TITLE
Add re-add deprecated support for unicode in a variety of APIs

### PR DESCRIPTION
**Pull request by exarkun**
Update of #189

### CODE REVIEW

**Comment by alex** [comment]
This is merge conflict'd because of a previous patch.

File: OpenSSL/test/test_ssl.py [hunk 1]
```diff
@@ -1,3 +1,5 @@
+
+
```
Comment (hynek) [inline OpenSSL/test/test_ssl.py]: these seem like an accident :)

File: OpenSSL/test/test_ssl.py [hunk 2]
```diff
@@ -2295,6 +2318,25 @@ def test_short(self):
         self.assertEquals(client.recv(1), b('x'))
 
 
+    def test_text(self):
+        """
+        :py:obj:`Connection.sendall` transmits all the content in the string passed to
+        it raising a DepreactionWarning in case of this being a text.
```
Comment (hynek) [inline OpenSSL/test/test_ssl.py]: `Depre*a*ctionWarning`

**Comment by hynek** [comment]
LGTM save the small things I pointed out.

File: OpenSSL/test/test_crypto.py [hunk 3]
```diff
@@ -13,7 +13,9 @@
 from subprocess import PIPE, Popen
 from datetime import datetime, timedelta
 
-from six import u, b, binary_type
+from six import u, b, binary_type, PY3
+from warnings import simplefilter
```
Comment (hynek) [inline OpenSSL/test/test_crypto.py]: warnings is a stdlib module therefore the import should be in the previous block

File: OpenSSL/test/test_ssl.py [hunk 4]
```diff
@@ -7,12 +7,13 @@
...
 from gc import collect, get_referrers
 from errno import ECONNREFUSED, EINPROGRESS, EWOULDBLOCK, EPIPE, ESHUTDOWN
-from sys import platform, version_info, getfilesystemencoding
+from sys import platform, getfilesystemencoding
 from socket import SHUT_RDWR, error, socket
 from os import makedirs
 from os.path import join
 from unittest import main
 from weakref import ref
+from warnings import catch_warnings, simplefilter
```
Comment (hynek) [inline OpenSSL/test/test_ssl.py]: wa comes before we :)

Comment (alex) [inline OpenSSL/test/test_ssl.py]: If we want to enforce this automatically, we can `pip install flake8-import-order` and run that in the tests.

Comment (exarkun) [inline OpenSSL/test/test_ssl.py]: Please continue with the current standard of importing modules in the order which minimizes heap allocation on Python 2.6.5 and Python 2.7.9.

Comment (alex) [inline OpenSSL/test/test_ssl.py]: That's a joke right?

Comment (exarkun) [inline OpenSSL/test/test_ssl.py]: Yes!  :fireworks: :balloon: :balloon:

**Pull request merged**